### PR TITLE
fix: notification email  message

### PR DESCRIPTION
**Pull request by jungwire**
### Description:

Fixes some issues with the notification email message:
 - Change sender name to 'BBMRI-ERIC negotiator'
 - List of Negotiations shows title instead of id
 - Link to Negotiation has appropriate role set

### Checklist:

_Make sure you tick all the boxes bellow if they are true or do not apply:_

- [x] I have performed a self review of my code
- [x] My code follows [Google Java Code style](https://google.github.io/styleguide/javaguide.html)
- [ ] I have made my code as simple as possible
- [ ] I have added unit tests and the code coverage has not decreased
- [ ] I have updated the documentation in all relevant places

### CODE REVIEW

File: src/main/java/eu/bbmri_eric/negotiator/service/UserNotificationServiceImpl.java [hunk 1]
```diff
@@ -263,14 +258,44 @@ private void sendEmail(@NonNull Person recipient, @NonNull List<Notification> no
...
             .collect(Collectors.toList());
 
+    Map<String, String> roleForNegotiation = new HashMap<>();
+
+    // Iterate over notifications to populate the map
+    for (Notification notification : notifications) {
+      String negotiationId = notification.getNegotiation().getId();
+      String role = extractRoleFromNotificationMessage(notification);
+      roleForNegotiation.put(negotiationId, role);
+    }
```
Comment (RadovanTomik) [inline src/main/java/eu/bbmri_eric/negotiator/service/UserNotificationServiceImpl.java:268]: put this in an appropriately named private method to remove the need for a comment

File: src/main/java/eu/bbmri_eric/negotiator/service/UserNotificationServiceImpl.java [hunk 2]
```diff
@@ -231,7 +226,7 @@ private Notification buildNewNotification(
   }
 
   @Override
-  @Scheduled(cron = "0 0 * * * *")
+  @Scheduled(cron = "0 */1 * ? * *")
```
Comment (RadovanTomik) [inline src/main/java/eu/bbmri_eric/negotiator/service/UserNotificationServiceImpl.java:229]: Can you please add this as a configuration?

File: src/main/java/eu/bbmri_eric/negotiator/service/UserNotificationServiceImpl.java [hunk 3]
```diff
@@ -263,14 +258,44 @@ private void sendEmail(@NonNull Person recipient, @NonNull List<Notification> no
...
+      String negotiationCreator = notification.getNegotiation().getCreatedBy().getName();
+      String postCreator = parts[1].trim();
+      return (negotiationCreator.equals(postCreator)) ? "ROLE_REPRESENTATIVE" : "ROLE_RESEARCHER";
+    } else if (message.matches("New Negotiation .*")) {
+      return "ROLE_REPRESENTATIVE";
+    } else {
+      return "ROLE_RESEARCHER"; // Default to ROLE_RESEARCHER
+    }
+  }
+
```
Comment (RadovanTomik) [inline src/main/java/eu/bbmri_eric/negotiator/service/UserNotificationServiceImpl.java:298]: maybe use switch statement? And feel free to remove the comments as the logic should be clear from the code.

**Review by RadovanTomik — CHANGES_REQUESTED** [review]

File: src/main/java/eu/bbmri_eric/negotiator/service/UserNotificationServiceImpl.java (hunk 1 above)
Comment (jungwire) [reply src/main/java/eu/bbmri_eric/negotiator/service/UserNotificationServiceImpl.java]: done

File: src/main/java/eu/bbmri_eric/negotiator/service/UserNotificationServiceImpl.java (hunk 2 above)
Comment (jungwire) [reply src/main/java/eu/bbmri_eric/negotiator/service/UserNotificationServiceImpl.java]: done

**Review by RadovanTomik — APPROVED** [review]

**Pull request merged**